src/Functions/ReplaceStringImpl.h

```diff
@@ -39,7 +39,11 @@ struct ReplaceStringImpl
         size_t input_rows_count)
     {
         if (needle.empty())
-            throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND, "Length of the pattern argument in function {} must be greater than 0.", name);
+        {
+            res_data.assign(haystack_data.begin(), haystack_data.end());
+            res_offsets.assign(haystack_offsets.begin(), haystack_offsets.end());
```

Comment:
`res_data = haystack_data` looks more simple

Reply:
/work/bigock/src/Functions/ReplaceStringImpl.h:43:22: error: object of type 'ColumnString::Chars' (aka 'PODArray<char8_t, 4096UL, Allocator<false, false>, PADDING_FOR_SIMD - 1, PADDING_FOR_SIMD>') cannot be assigned because its copy assignment operator is implicitly deleted
   43 |             res_data = haystack_data;